compiler/noirc_frontend/src/hir/def_collector/dc_mod.rs

```diff
@@ -643,13 +672,14 @@ impl<'a> ModCollector<'a> {
     /// On error this returns None and pushes to `errors`
     fn push_child_module(
         &mut self,
+        context: &mut Context,
         mod_name: &Ident,
-        file_id: FileId,
+        mod_location: Location,
         add_to_parent_scope: bool,
         is_contract: bool,
-    ) -> Result<LocalModuleId, DefCollectorErrorKind> {
+    ) -> Result<ModuleId, DefCollectorErrorKind> {
         let parent = Some(self.module_id);
-        let location = Location::new(mod_name.span(), file_id);
+        let location = Location::new(mod_name.span(), mod_location.file);
```

Comment:
It looks like we have a lot of overlap between `mod_name`, `mod_location` and `location`. Do we really need all three of these? Seems like we can remove one or the other of `mod_location` and `location` at least. (maybe so for structs/submodules but not modules in a separate file?)

Reply:
> I don't know what that part of contracts is used for.

This is use to display codelenses above the `contract Foo` line so we're going to want to maintain this.

I'm inclined to just make a note of this so we can merge these PRs and address this as potential tech debt in the future.

Reply:
I'll add a note 👍